cmd/helm/options.go

```diff
@@ -0,0 +1,227 @@
+func readFile(filePath string) ([]byte, error) {
+	if strings.TrimSpace(filePath) == "-" {
+		return ioutil.ReadAll(os.Stdin)
+	}
+	u, _ := url.Parse(filePath)
+	p := getter.All(settings)
+
+	// FIXME: maybe someone handle other protocols like ftp.
+	getterConstructor, err := p.ByScheme(u.Scheme)
+
+	if err != nil {
+		return ioutil.ReadFile(filePath)
+	}
+
+	getter, err := getterConstructor(filePath, "", "", "")
+	if err != nil {
+		return []byte{}, err
+	}
+	data, err := getter.Get(filePath)
+	return data.Bytes(), err
+}
+
+// -----------------------------------------------------------------------------
+// Chart Path Options
+
+type chartPathOptions struct {
+	caFile   string // --ca-file
+	certFile string // --cert-file
+	keyFile  string // --key-file
+	keyring  string // --keyring
+	password string // --password
+	repoURL  string // --repo
+	username string // --username
+	verify   bool   // --verify
+	version  string // --version
+}
+
+// defaultKeyring returns the expanded path to the default keyring.
+func defaultKeyring() string {
+	if v, ok := os.LookupEnv("GNUPGHOME"); ok {
+		return filepath.Join(v, "pubring.gpg")
+	}
+	return os.ExpandEnv("$HOME/.gnupg/pubring.gpg")
+}
+
+func (o *chartPathOptions) addFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.version, "version", "", "specify the exact chart version to install. If this is not specified, the latest version is installed")
+	fs.BoolVar(&o.verify, "verify", false, "verify the package before installing it")
+	fs.StringVar(&o.keyring, "keyring", defaultKeyring(), "location of public keys used for verification")
+	fs.StringVar(&o.repoURL, "repo", "", "chart repository url where to locate the requested chart")
+	fs.StringVar(&o.username, "username", "", "chart repository username where to locate the requested chart")
+	fs.StringVar(&o.password, "password", "", "chart repository password where to locate the requested chart")
+	fs.StringVar(&o.certFile, "cert-file", "", "identify HTTPS client using this SSL certificate file")
+	fs.StringVar(&o.keyFile, "key-file", "", "identify HTTPS client using this SSL key file")
+	fs.StringVar(&o.caFile, "ca-file", "", "verify certificates of HTTPS-enabled servers using this CA bundle")
+}
+
+func (o *chartPathOptions) locateChart(name string) (string, error) {
+	return locateChartPath2(o.repoURL, o.username, o.password, name, o.version, o.keyring, o.certFile, o.keyFile, o.caFile, o.verify)
+}
+
+// locateChartPath looks for a chart directory in known places, and returns either the full path or an error.
+//
+// This does not ensure that the chart is well-formed; only that the requested filename exists.
+//
+// Order of resolution:
+// - relative to current working directory
+// - if path is absolute or begins with '.', error out here
+// - chart repos in $HELM_HOME
+// - URL
+//
+// If 'verify' is true, this will attempt to also verify the chart.
+func locateChartPath2(repoURL, username, password, name, version, keyring,
```

Comment:
Update comment to have this name in it?

Reply:
It looks like locateChartPath got removed. Is there a reason to call this locateChartPath2?